### PR TITLE
Ubports build

### PR DESCRIPTION
**Pull request by jonnius**
Build instructions can be found in **packaging/ubports/README**. In short:

* install [Clickable](http://clickable.bhdouglass.com/en/latest/install.html)
* run `ln -s packaging/ubports/clickable.json clickable.json`
* run `clickable prepare-deps` (to download all sources)
* run `clickable build-libs --arch amd64`
* run `clickable desktop --dirty` (the `--dirty` prevents Clickable from cleaning the build dir each time)

The window won't even show up in desktop mode. On the phone it gets stuck at the app splash screen.

### CODE REVIEW

File: build-libpostal.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+./bootstrap.sh
+
+ARCH_FLAGS=""
+case ${ARCH_TRIPLET} in
+"arm-linux-gnueabihf" )
+    ARCH_FLAGS="--with-cflags-scanner-extra=-marm"
+    ;;
+"aarch64-linux-gnu" )
+    # TODO what do we need to pass here for arm64?
+    ARCH_FLAGS="--with-cflags-scanner-extra=-marm"
```
Comment (jonnius) [inline build-libpostal.sh]: Any idea which flag is needed for arm64 builds? It does neither work with nor without `-marm`.

Comment (jonnius) [reply build-libpostal.sh]: Ok, somehow [managed](https://github.com/openvenues/libpostal/issues/349#issuecomment-575805153) to make it work.

File: prepare-deps.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+LIBPOSTAL_SRC_DIR="${ROOT}/libs/libpostal"
+MAPNIK_SRC_DIR="${ROOT}/libs/mapnik"
+VALHALLA_SRC_DIR="${ROOT}/libs/valhalla"
+
+# Remove old downloads
+rm -rf "${LIBOSMSCOUT_SRC_DIR}" "${LIBPOSTAL_SRC_DIR}" "${MAPNIK_SRC_DIR}" "${VALHALLA_SRC_DIR}"
+
+# Download sources
+git clone --recursive --shallow-submodules --depth 1 https://github.com/rinigus/libosmscout.git "${LIBOSMSCOUT_SRC_DIR}"
+git clone --recursive --shallow-submodules -b devel https://github.com/rinigus/pkg-libpostal.git libs/libpostal "${LIBPOSTAL_SRC_DIR}"
```
Comment (rinigus) [inline prepare-deps.sh]: `libs/libpostal` should be removed as we have destination specified later in command line. as it is, it fails to clone.

File: prepare-deps.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+git clone --recursive -b master https://github.com/rinigus/pkg-mapnik "${MAPNIK_SRC_DIR}"
+git clone --recursive -b master --shallow-submodules https://github.com/rinigus/pkg-valhalla-lite.git "${VALHALLA_SRC_DIR}"
```
Comment (rinigus) [inline prepare-deps.sh]: I would suggest to drop `--shallow-submodules` here. for some reason it didn't work for me.

**Comment by rinigus** [comment]
Mapnik fails for me on amd64 with 

```
  File "/usr/lib/python2.7/os.py", line 150:
    makedirs(head, mode)
  File "/usr/lib/python2.7/os.py", line 157:
    mkdir(name, mode)
make: *** [install] Error 2
Makefile:17: recipe for target 'install' failed
Command exited with non-zero exit status 2, see above for details. This is most likely not a problem with Clickable.
```

**Comment by rinigus** [comment]
When removing Mapnik section in clickable, after `clickable build-libs --arch amd64`, `clickable desktop --dirty` fails for me with 

```
Project ERROR: libvalhalla development package not found
Command exited with non-zero exit status 3, see above for details. This is most likely not a problem with Clickable.
```

Please check with the clean build if it works for you. Will wait now for your feedback before proceeding

File: prepare-deps.sh (hunk 2 above)
Comment (jonnius) [reply prepare-deps.sh]: Done.

**Comment by jonnius** [comment]
> In addition, please move all packaging scripts to `packaging/ubports` as we did for Pure Maps.

Do

File: prepare-deps.sh (hunk 3 above)
Comment (jonnius) [reply prepare-deps.sh]: Done

**Comment by jonnius** [comment]
I just cleaned up and tried it myself by starting over clean. Should be fine now.

I forgot to mention that the last step requires latest Clickable from master branch (which you get via pip). This morning a patch was merged that we need.

**Comment by rinigus** [comment]
Thank you very much, now managed to build. So, ready for porting...

**Comment by rinigus** [comment]
Its not fully working yet. I have to know where in the tree inside clickable is the app installed. Currently PREFIX is not set. As a result, Valhalla config and other data files are not accessible

**Comment by jonnius** [comment]
Each library has its own install dir, which is `build/x86_64-linux-gnu/valhalla/install` for example for valhalla and amd64. During valhalla build this path is available as env var `INSTALL_DIR`.

Files and directories from library builds that need to be shipped with the app need to be specified explicitly, e.g. with [install_data](http://clickable.bhdouglass.com/en/latest/clickable-json.html#install-data). You can use `valhalla_LIB_INSTALL_DIR` to access valhalla's install dir.

**Comment by jonnius** [comment]
Btw. Clickable's `cmake` template does `-DCMAKE_INSTALL_PREFIX:PATH=/.` (which is the build dir).

**Comment by rinigus** [comment]
Not sure I understand how to proceed. We need to specify where `$${PREFIX}/share/$${TARGET}/data/valhalla.json` goes and where can it be found when application runs in the clickable.

Currently, PREFIX is not set and we use `/usr/local` for it. Where it will not find anything on the phone.

**Comment by jonnius** [comment]
Good to merge now.

**Pull request merged**